projection/nft/nft.go

```diff
@@ -0,0 +1,321 @@
+			}
+
+			tokenRow := view.TokenRow{
+				DenomId:      msgBurnNFT.DenomId,
+				TokenId:      msgBurnNFT.TokenId,
+				Drop:         prevTokenRow.Drop,
+				Burned:       true,
+				Name:         prevTokenRow.Name,
+				URI:          prevTokenRow.URI,
+				Data:         prevTokenRow.Data,
+				Minter:       prevTokenRow.Minter,
+				Owner:        prevTokenRow.Owner,
+				MintedAt:     prevTokenRow.MintedAt,
+				LastEditedAt: blockTime,
+			}
+
+			if updateTokenErr := tokensView.Update(tokenRow); updateTokenErr != nil {
+				return fmt.Errorf("error updating NFT token row: %v", updateTokenErr)
+			}
+
+			if decrementErr := tokensTotalView.DecrementAll([]string{
+				fmt.Sprintf("-:-:-:%s", tokenRow.Owner),
+				fmt.Sprintf("%s:-:-:%s", tokenRow.DenomId, tokenRow.Owner),
+				fmt.Sprintf("-:%s:-:%s", tokenRow.Drop, tokenRow.Owner),
+				fmt.Sprintf("-:-:%s:%s", tokenRow.Minter, tokenRow.Owner),
+				fmt.Sprintf("%s:%s:-:%s", tokenRow.DenomId, tokenRow.Drop, tokenRow.Owner),
+				fmt.Sprintf("%s:-:%s:%s", tokenRow.DenomId, tokenRow.Minter, tokenRow.Owner),
+				fmt.Sprintf("-:%s:%s:%s", tokenRow.Drop, tokenRow.Minter, tokenRow.Owner),
+				fmt.Sprintf("%s:%s:%s:%s", tokenRow.DenomId, tokenRow.Drop, tokenRow.Minter, tokenRow.Owner),
+			}, 1); decrementErr != nil {
+				return fmt.Errorf("error decrementing NFT token total: %v", decrementErr)
+			}
+
+		} else if msgTransferNFT, ok := event.(*event_usecase.MsgNFTTransferNFT); ok {
+			prevTokenRow, queryPrevTokenRow := tokensView.FindById(msgTransferNFT.DenomId, msgTransferNFT.TokenId)
+			if queryPrevTokenRow != nil {
+				return fmt.Errorf("error querying NFT token being edited: %v", queryPrevTokenRow)
+			}
+
+			transferRow := view.TokenTransferRow{
+				DenomId:         msgTransferNFT.DenomId,
+				TokenId:         msgTransferNFT.TokenId,
+				Drop:            prevTokenRow.Drop,
+				BlockHeight:     height,
+				TransactionHash: msgTransferNFT.TxHash(),
+				Sender:          msgTransferNFT.Sender,
+				Recipient:       msgTransferNFT.Recipient,
+				TransferredAt:   blockTime,
+			}
+
+			if insertTransferErr := tokenTransfersView.Insert(transferRow); insertTransferErr != nil {
+				return fmt.Errorf("error inserting NFT token transfer into view: %v", insertTransferErr)
+			}
+		}
+	}
+
+	if err := nft.UpdateLastHandledEventHeight(rdbTxHandle, height); err != nil {
+		return fmt.Errorf("error updating last handled event height: %v", err)
+	}
+
+	if err := rdbTx.Commit(); err != nil {
+		return fmt.Errorf("error committing changes: %v", err)
+	}
+	committed = true
+	return nil
+}
+
+func (nft *NFT) updateToken(tokensView *view.Tokens, tokenRow view.TokenRow, tokensTotalView *view.TokensTotal) error {
+	if updateTokenErr := tokensView.Update(tokenRow); updateTokenErr != nil {
+		return fmt.Errorf("error updating NFT token row: %v", updateTokenErr)
+	}
+
+	if decrementErr := tokensTotalView.DecrementAll([]string{
```

Comment:
What's happening here, we are decrementing and then incrementing `tokensTotalView` by 1.
Sorry, Could you brief me?

Reply:
Good catch, it is trying to update previous owner's entry then increment the new one. Code updated.